plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2576,8 +2576,7 @@ public void validateScan(ConnectorSession session, ConnectorTableHandle tableHan
         // all references are simple variables
         if (columnProjections.values().stream().allMatch(ProjectedColumnRepresentation::isVariable)) {
             Set<ColumnHandle> projectedColumns = ImmutableSet.copyOf(assignments.values());
-            if (hiveTableHandle.getProjectedColumns().isPresent()
-                    && hiveTableHandle.getProjectedColumns().get().equals(projectedColumns)) {
+            if (hiveTableHandle.getProjectedColumns().equals(projectedColumns)) {
```

Comment:
There is a diffrence between Optional.empty and empty set.
Not sure it really matters though.

if it does, we would probably want to change IcebergTableHanld.ProjectedColumns

cc @alexjo2144 @losipiuk @phd3